src/opensearchDatasource.ts

```diff
@@ -662,6 +678,103 @@ export class OpenSearchDatasource
     );
   }
 
+  runLiveQueryThroughBackend(request: DataQueryRequest<OpenSearchQuery>): Observable<DataQueryResponse> {
+    const liveQueries = request.targets.filter((query) => {
+      return query.metrics?.length === 1 && query.metrics[0].type === 'logs' && !query.hide;
+    });
+
+    if (liveQueries.length === 0) {
+      console.log('[OpenSearch runLiveQueryThroughBackend] No live stream-able queries found.');
+      return of({ data: [], state: LoadingState.Done, key: request.requestId });
+    }
+
+    const firstLiveQuery = liveQueries[0];
+    if (!firstLiveQuery) {
+      return of({ data: [], state: LoadingState.Done, key: request.requestId });
+    }
+
+    const { format, ...streamingQueryPayload } = firstLiveQuery;
+    const finalStreamingQuery = {
+      ...streamingQueryPayload,
+      query: streamingQueryPayload.query || '*',
+      refId: firstLiveQuery.refId,
+    };
+
+    const streamPath = `tail/${firstLiveQuery.refId}`;
```

Comment:
I think we can improve the stream path here:
streaming should be reusable - if one user is streaming a query from OpenSearchDS_123 and another user is requesting the same query, this stream should be reused. SO our stream path should reflect this - it can be done same as Loki [here](https://github.com/grafana/grafana/blob/ac832c157e12f6c7c65f74f73d5ea92541eb7d8f/public/app/plugins/datasource/loki/streaming.ts#L22) This means it doesn't need the key either, since the stream path functions as the key later in stream_handler (instead of refid)
Does that make sense?